vowpalwabbit/parse_args.cc

```diff
@@ -1631,7 +1409,15 @@ void parse_modules(
 
   parse_output_preds(options, all);
 
-  parse_reductions(options, all);
+  // create reduction stack builder instance
+  auto learner_builder = VW::make_unique<VW::default_reduction_stack_setup>(all);
+  // kick-off reduction setup functions
+  all.l = learner_builder->operator()(options, all);
+
+  // explicit destroy of learner_builder state
+  // avoids misuse of this interface:
+  learner_builder.reset();
```

Comment:
added this as a 'canary in the coalmine' for future PRs that expect learner_builder instance that is being passed thru setup reduction functions to be a long-lived one, when it should not be the case.

Reply:
Good idea.
We can revisit this later if we need some sort of clone & slice functionality for things like automl in case we want, for example, to have coin as a challenger to SGD. I currently have no idea on how that would it look like.